src/structures.rs

```diff
@@ -23,26 +23,107 @@ impl<'a> From<&'a gtfs_structures::Stop> for Stop {
     }
 }
 
-struct Connection {
-    trip: u32,
-    dep_time: u16,
-    arr_time: u16,
-    dep_stop: u32,
-    arr_stop: u32,
+impl Stop {
+    pub fn new(id: &str) -> Self {
+        Self {
+            id: id.to_owned(),
+            name: id.to_owned(),
+            parent_station: None,
+            location_type: gtfs_structures::LocationType::StopPoint,
+        }
+    }
+}
+
+pub struct Connection {
+    pub trip: usize,
+    pub dep_time: u16,
+    pub arr_time: u16,
+    pub dep_stop: usize,
+    pub arr_stop: usize,
```

Comment:
usize will not be costly? Strange when we see your u16 choice for date time.

Reply:
it will. But then I thought that I should stop early optimization. I’m yet very far from any performance considerations. Let’s keep it simple for now, and change it back later.